File: backend/models/finance.py
from datetime import date
from typing import List, Literal, Optional

from pydantic import BaseModel


class Persona(BaseModel):
    id: str
    name: str
    description: str


class TransactionRecord(BaseModel):
    persona_id: str
    date: date
    description: Optional[str] = None
    category: str
    amount: float
    type: Literal["income", "expense"]
    essential: bool = False


class MonthlyOverview(BaseModel):
    month: str
    total: float
    income: float
    savings: float


class CategorySummary(BaseModel):
    name: str
    latest: float
    essential: bool


class GoalsSummary(BaseModel):
    target_savings_rate: float
    current_savings_rate: float


class FinanceSummary(BaseModel):
    monthly_overview: List[MonthlyOverview]
    categories: List[CategorySummary]
    goals: GoalsSummary

File: frontend/src/types/finance.ts
export interface Persona {
  id: string;
  name: string;
  description: string;
}

export interface MonthlyOverview {
  month: string; // YYYY-MM
  total: number;
  income: number;
  savings: number;
}

export interface CategorySummary {
  name: string;
  latest: number;
  essential: boolean;
}

export interface FinanceSummary {
  monthly_overview: MonthlyOverview[];
  categories: CategorySummary[];
  goals: {
    target_savings_rate: number;
    current_savings_rate: number;
  };
}

export interface ChatMessage {
  id: string;
  role: 'user' | 'assistant' | 'system';
  content: string;
}

export interface ChatMetadata {
  model: string;
  latency_ms: number;
}

export interface ChatResponse {
  message: ChatMessage;
  metadata: ChatMetadata;
}
